odoo/tools/translate.py

```diff
@@ -521,7 +521,16 @@ def __call__(self, source, *args, **kwargs):
     def _get_translation(self, source, module=None):
         try:
             frame = inspect.currentframe().f_back.f_back
-            lang = self._get_lang(frame)
+            translation_file = inspect.getfile(frame)
+            lang = None
+            while True:
+                lang = self._get_lang(frame)
```

Comment:
I think already trying to find the `lang` here defeats a bit the purpose of what Chong suggested, to find really a specific `context` variable in the frame's local variables. The `_get_lang()` function tries a lot of things and in the end falls back on the default language `en_US`, by the method `env['res.users'].context_get()['lang']` if it really can't find anything else.

A test would be helpful, so you can see whether it really works.

Reply:
Done!